### ISSUE TITLE
Review accessibility of Clay React components

### ISSUE THREAD

**Issue by diegonvs** [liferay/clay#4762]
## What is your proposal?

We should evaluate our components to see what we're missing in terms of a11y as well as what the developer can do in the form of a well-documented guide for using a certain component in an accessible manner.

There is some useful information that we may iterate through to examine our accessibility:

* **Bootstrap Documentation**:

Each BS component has its own a11y guide, and because the structure of BS is similar to that of Clay, we can revisit each component and see what we can learn. Consider the following scenario: See this [documentation](https://getbootstrap.com/docs/5.1/components/dropdowns/#accessibility), which we can use as an example of what is required from an accessibility standpoint for a dropdown and decide whether it makes sense to implement or document on our side.

* **Examples of WAI-ARIA**:

We may take each example [on this page](https://www.w3.org/TR/wai-aria-practices/examples/) and see whether we have the same component pattern on Clay's side, and if this accessibility patch should be on our side or if we should provide a developer guide.

## Why would adopting this proposal be beneficial?

Increase accessibility for our React components

## What are the alternatives to this proposal?

I don't know 😢

**Comment by matuzalemsteles** [comment]
Closing this out as in the last few months we've made a lot of accessibility improvements to our components and we've been focusing on that since last year.
